### PR TITLE
feat: add build_info metric

### PR DESCRIPTION
**Pull request by t3mi**
- Adds build_info prometheus metric

### CODE REVIEW

**Comment by jmos5156** [comment]
Can we review this change and get it in ASAP?
We are in need of this fix to be able to move forward

**Comment by t3mi** [comment]
@mustafaakin any chance for review?

**Comment by smelchior** [comment]
We would also very much like to get this merged, as we are loosing events currently.

**Comment by jmos5156** [comment]
Is there any other way to get this into master? Any other reviewers?
I think the general consensus is that this is a good thing to include.
I'm not a Golang developer but the change that fits my needs seems to be ok.

**Comment by mustafaakin** [comment]
We can merge this, however it has 2 features. The build info is OK, great. However, the calling update on every update should be optional otherwise all the new releases will get substantially larger amount of events. If you can separate this, I'd be glad to merge the build info. 

Additionally, go 1.18 introduced a new package for natively accessing build information. https://shibumi.dev/posts/go-18-feature/ It'd be great to use it.

File: pkg/kube/watcher.go [hunk 1]
```diff
@@ -57,7 +57,8 @@ func (e *EventWatcher) OnAdd(obj interface{}) {
 }
 
 func (e *EventWatcher) OnUpdate(oldObj, newObj interface{}) {
-	// Ignore updates
+	event := newObj.(*corev1.Event)
+	e.onEvent(event)
```
Comment (mustafaakin) [inline pkg/kube/watcher.go:61]: This should be on other update, carefully thought. Not all sinks are compatible with this kind of update flow.

**Review by mustafaakin — CHANGES_REQUESTED** [review]

File: pkg/kube/watcher.go (hunk 1 above)
Comment (t3mi) [reply pkg/kube/watcher.go]: I just tried to restore what was before. Anyway, there is already an open PR for such configurable functionality #126.

@mustafaakin please review.

**Comment by jmos5156** [comment]
It feels like this is dead in the water and not going anywhere. If so could we abandon this, it will allow me to look at other alternatives seeing as this repo is not really responsive to contributions and making the product better.

**Review by mustafaakin — APPROVED** [review]

**Pull request merged**